### ISSUE TITLE
Breadcrumbs API

### ISSUE THREAD

**Issue by joshuap** [honeybadger-io/honeybadger-crystal#6]
The client should have a **public** `add_breadcrumb()` method that stores an array of breadcrumbs. There should also be a public `clear()` method (see also: #4) to reset the breadcrumbs to the default (`[]`).

- Data should be stored [local to the current thread](https://www.notion.so/Client-Library-Spec-aa891332f7874196aa0695b6d38dca66)
- Data should be sanitized, but not [filtered](https://www.notion.so/Client-Library-Spec-aa891332f7874196aa0695b6d38dca66)

**Supported config options:**

- `breadcrumbs_enabled`
    - optional
    - default: `true`
- `max_breadcrumbs`
    - optional
    - default: `40`

**Example:**

```ruby
# Supported configuration options w/ defaults:
Honeybadger.configure({
  # optional
  breadcrumbs_enabled: true,

  # optional
  # The max number of breadcrumbs to store at one time.
  # The oldest breadcrumbs should be dropped as newer breadcrumbs are added.
	max_breadcrumbs: 40
})

# Adding a breadcrumb:
Honeybadger.add_breadcrumb(
	"Email Sent",
	metadata: {
		user: user.id,
		message: message
	}
)

# Clearing context and breadcrumbs:
Honeybadger.clear()
```

### Resources
- [Client Library Spec](https://www.notion.so/honeybadger/Client-Library-Spec-aa891332f7874196aa0695b6d38dca66#d9a1892930694580a9f41eebb99c22bd)
- [Honeybadger Documentation](https://docs.honeybadger.io)
- [Exceptions API Documentation](https://docs.honeybadger.io/api/exceptions.html)
- [Notices JSON Schema](https://github.com/honeybadger-io/honeybadger-schemas/blob/master/exceptions/v1/notices.json)

**Comment by jgaskins** [comment]
With what I'm doing with events in #37, I was thinking I might also take a swing at this one since events and breadcrumbs seem pretty similar. Is the client library spec link still accurate? If so, it appears not to be public.

**Comment by subzero10** [comment]
> I was thinking I might also take a swing at this one since events and breadcrumbs seem pretty similar.

Indeed, they are similar in the way that they are queued locally and sent to the API in batches.
However, there are a couple of things to clarify here (probably you are already aware, but I guess there's no harm repeating just in case 😄):
1. Breadcrumbs are sent along with an error/notice. Events will have to be sent either when a threshold in terms of size or time has passed (as you are already doing in #37).
2. There's a max number of breadcrumbs stored in the local queue. If the number is exceeded, the older breadcrumbs are dropped. This is not the case for the Events API.

> Is the client library spec link still accurate? If so, it appears not to be public.

Hey @jgaskins, the client library spec [link](https://honeybadger.notion.site/Client-Library-Spec-aa891332f7874196aa0695b6d38dca66) should be OK now.
